subsys/bluetooth/host/scan.c

```diff
@@ -14,6 +14,8 @@
 #include <zephyr/bluetooth/buf.h>
 #include <zephyr/bluetooth/direction.h>
 #include <zephyr/bluetooth/addr.h>
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/hci_vs.h>
```

Comment:
I'm not super happy that we include those now, people might be tempted to use more HCI stuff now that they're here.

Reply:
This file is a Host internal. Is it an issue that there is direct use of HCI header files?
There is no separation layer that covers HCI interface and its types.